runner/database_neo4j.go

```diff
@@ -0,0 +1,57 @@
+package runner
+
+import (
+	"io"
+
+	jsonutil "github.com/multiprocessio/go-json"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+func (ec EvalContext) evalNeo4j(panel *PanelInfo, dbInfo DatabaseConnectorInfoDatabase, server *ServerInfo, w io.Writer) error {
+	_, conn, err := ec.getConnectionString(dbInfo)
+	if err != nil {
+		return err
+	}
+
+	password, err := ec.decrypt(&dbInfo.Password)
+	if err != nil {
+		return err
+	}
+
+	driver, err := neo4j.NewDriver(conn, neo4j.BasicAuth(dbInfo.Username, password, ""))
+	if err != nil {
+		return err
+	}
+	defer driver.Close()
+
+	sess := driver.NewSession(neo4j.SessionConfig{})
+
+	result, err := sess.Run(panel.Content, nil)
+	if err != nil {
+		return err
+	}
+
+	return withJSONArrayOutWriterFile(w, func(w *jsonutil.StreamEncoder) error {
+		records, err := result.Collect()
+		if err != nil {
+			return nil
+		}
+
+		for _, record := range records {
+			row := make(map[string]any)
```

Comment:
Allocations inside a loop are not a good idea. They take too much time and are a function of the size of the output as you have it.

Instead outside of the for loop do:

```
row := map[string]any{}
```

(Also I prefer to avoid `make` except where you have to like channels.)

The only thing to worry about after that is that every row must include all keys otherwise you may include the previous row's results like this bug: https://github.com/multiprocessio/datastation/issues/217.